backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/query/impl/LuceneSearchScrollImpl.java

```diff
@@ -88,21 +89,24 @@ public void close() {
 			if ( search != null ) {
```

Comment:
I believe the condition `scrollIndex + chunkSize > queryFetchSize` needs to be updated? It would be `( scrollIndex + chunkSize ) > ( offset + queryFetchSize )`.

Maybe we need an additional test, because something should have failed...

This is getting very confusing... I suggest the following renamings:

* `search` => `currentPage`
* `offset` => `currentPageOffset`
* `queryFetchSize` => `currentPageLimit`
* `index` => `currentChunkStartIndexInPage`
* `endIndexExclusive` => `currentChunkEndIndexInPage`
* `scrollIndex` => `nextChunkOffset`

The condition would become:

```
( nextChunkOffset + chunkSize ) > ( currentPageOffset + currentPageLimit )
```

Reply:
you're right... going to fix it. Thanks

Reply:
Done. About the additional test, I think that we don't need it. 
Because the external behavior would be the same and what we change with your fix is to avoid doing more scroll works than the ones we really need.